internal/system/agent/services.go

```diff
@@ -51,117 +50,220 @@ func InvokeOperation(action string, services []string) bool {
 				err := stopper.Stop(service)
 				if err != nil {
 					msg := fmt.Sprintf("error stopping service \"%s\": %v", service, err)
-					logs.LoggingClient.Error(msg)
+					LoggingClient.Error(msg)
 				}
 			} else {
 				msg := fmt.Sprintf("stopping not supported with specified executor")
-				logs.LoggingClient.Warn(msg)
+				LoggingClient.Warn(msg)
 			}
 
 		case RESTART:
 			if restarter, ok := executorClient.(interfaces.ServiceRestarter); ok {
 				err := restarter.Restart(service)
 				if err != nil {
 					msg := fmt.Sprintf("error restarting service \"%s\": %v", service, err)
-					logs.LoggingClient.Error(msg)
+					LoggingClient.Error(msg)
 				}
 			} else {
 				msg := fmt.Sprintf("restarting not supported with specified executor")
-				logs.LoggingClient.Warn(msg)
+				LoggingClient.Warn(msg)
 			}
 		}
 	}
 	return true
 }
 
-func getConfig(services []string, ctx context.Context) (map[string]interface{}, error) {
+func getConfig(services []string, ctx context.Context) (ConfigRespMap, error) {
 
-	configuration := make(map[string]interface{})
+	c := ConfigRespMap{}
+	c.Configuration = map[string]interface{}{}
 
-	// Loop through requested actions, along with respectively-supplied parameters.
+	// Loop through requested actions, along with (any) respectively-supplied parameters.
 	for _, service := range services {
 
+		// Check whether SMA does _not_ know of ServiceKey ("service") as being one for one of its ready-made list of clients.
 		if !isKnownServiceKey(service) {
-			logs.LoggingClient.Warn("unknown service found getting configuration", "service name", service)
-		}
+			LoggingClient.Info(fmt.Sprintf("service %s not known to SMA as being in the ready-made list of clients", service))
 
-		responseJSON, err := clients[service].FetchConfiguration(ctx)
-		if err != nil {
-			configuration[service] = fmt.Sprintf("%s get config error: %s", service, err.Error())
-			logs.LoggingClient.Error("error retrieving configuration", "service name", service, "errMsg", err.Error())
+			// Service unknown to SMA, so ask the Registry whether `service` is available.
+			err := registryClient.IsServiceAvailable(service)
+			if err != nil {
+				c.Configuration[service] = fmt.Sprintf("Registry responded with %s service not available, with error code: %s", service, err.Error())
+				LoggingClient.Error(fmt.Sprintf(err.Error()))
+			} else {
+				LoggingClient.Info(fmt.Sprintf("Registry responded with %s service available", service))
+
+				// Since service is unknown to SMA, ask the Registry for a ServiceEndpoint associated with `service`
+				e, err := registryClient.GetServiceEndpoint(service)
+				if err != nil {
+					c.Configuration[service] = fmt.Sprintf("on attempting to get ServiceEndpoint for service %s, got error: %v", service, err)
+					LoggingClient.Error(err.Error())
+				} else {
+					// Preparing to add the specified key to the map where the value will be the respective GeneralClient
+					var clientInfo = config.ClientInfo{}
```

Comment:
Or just `clientInfo := config.ClientInfo{}`

Reply:
Yep, will make it so, as that's better aligned with `Go` idioms 👍